kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/descriptors/descriptor.kt

```diff
@@ -103,15 +103,15 @@ sealed interface Descriptor {
     */
    fun isParentOf(descriptor: Descriptor): Boolean = when (descriptor) {
       is SpecDescriptor -> false // nothing can be the parent of a spec
-      is TestDescriptor -> this.id == descriptor.parent.id
+      is TestDescriptor -> this == descriptor.parent
```

Comment:
This fails if two things on the same path have a parent name that is the same ?
Can we add an explicit test for this to the descriptor tests ?

Reply:
[Tests added](https://github.com/kotest/kotest/pull/3071/commits/b0230f0c43f29169b77bd33535b081e56104ed53)